src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -1100,24 +1100,44 @@ public bool ProcessTransaction(Transaction transaction)
         }
 
         /// <inheritdoc />
-        public void DeleteWallet()
+        public void DeleteWallet(string walletName)
         {
-            throw new NotImplementedException();
+            lock (this.lockObject)
+            {
+                // Back-up to JSON ".wallet.json.bak" first
+                this.SaveWallet(walletName, true);
+
+                // Delete any JSON wallet file that may otherwise be re-imported.
```

Comment:
Why do we need to delete JSON wallets? Seems like not that coherent approach given we are moving away from JSON. Could you perhaps consider leaving JSON files as is and just work on DB? This prob would allow people to resync JSON wallet by deleting database wallet. Just something to consider

Reply:
The purpose of the new method is to delete database wallets but we will not achieve much if the wallet gets re-imported from JSON the next time the node starts up. Hence we have to delete any JSON file found there too. The SaveWallet creates a backup in-case the user made an error and also allows the "deleted" wallet to be re-imported into a different repository.

Reply:
maybe json wallet that got "deleted" could just be moved to a folder where the user has the chance to restore them back if needed?

Reply:
Renaming basically achieves the same. The alternative is to move it to the wallets folder and not rename it.

Reply:
sounds good to me